test/step-function.test.ts

```diff
@@ -510,6 +510,88 @@ test("boolean logic", () => {
   expect(normalizeDefinition(definition)).toMatchSnapshot();
 });
 
+test.each([
+  ["with", undefined],
+  [
+    "without",
+    {
+      optimization: {
+        joinConsecutiveChoices: false,
+        optimizeVariableAssignments: false,
+        removeNoOpStates: false,
+        removeUnreachableStates: false,
+      },
+    } as ASLOptions,
+  ],
+])("%s optimizer", (_, opt) => {
```

Comment:
Love this test. 

Does this mean that all optimizations are turned on by default?

Reply:
yeah. For now at least.  

```ts
  export const DefaultOptimizeOptions: OptimizeOptions = {
    optimizeVariableAssignments: true,
    removeUnreachableStates: true,
    joinConsecutiveChoices: true,
    removeNoOpStates: true,
  };
```

I imagine we'll create new, experimental ones or versions of existing ones and make a breaking change to turn them on.

Reply:
> Love this test.

I am glad I wrote it, found that options in SFN were borked this whole time 🤒